master/internal/api_project.go

```diff
@@ -53,45 +53,179 @@ func (a *apiServer) getProjectColumnsByID(
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "start_time",
+			DisplayName: "Start time",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_DATE,
+		},
+		{
+			Column:      "duration",
+			DisplayName: "Duration",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER, // ?
+		},
+		{
+			Column:      "num_trials",
+			DisplayName: "Trial count",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "state",
+			DisplayName: "State",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "searcher_type",
+			DisplayName: "Searcher type",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "resource_pool",
+			DisplayName: "Resource pool",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
+		{
+			Column:      "progress",
+			DisplayName: "Progress",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "checkpoint_size",
+			DisplayName: "Checkpoint size",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "checkpoint_count",
+			DisplayName: "Checkpoint count",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_NUMBER,
+		},
+		{
+			Column:      "username",
+			DisplayName: "User",
+			Location:    projectv1.LocationType_LOCATION_TYPE_EXPERIMENT,
+			Type:        projectv1.ColumnType_COLUMN_TYPE_TEXT,
+		},
 	}
+
 	// Get hyperpatameters columns
-	hyperparameters := struct {
-		Hyperparameters []string
+	hyperparametersQuery := db.Bun().NewSelect().
+		ColumnExpr("jsonb_array_elements_text(hyperparameters) as hyperparameter").
+		Table("projects").
+		Where("id = ?", id)
+
+	trialsQuery := db.Bun().NewSelect().
+		Column("hparams").
+		Table("trials").
+		Join("experiments on trials.experiment_id = experiments.id").
+		Where("experiments.project_id = ?", id)
+
+	hyperParametersWithExamples := []struct {
+		Hyperparameter string
+		ExampleType    string
 	}{}
-	err := db.Bun().
-		NewSelect().Table("projects").Column("hyperparameters").Where(
-		"id = ?", id).Scan(ctx, &hyperparameters)
+
+	/**
+	 * Note -- this scales badly on the trials table. can we either
+	 *   - guarantee that all trials for an experiment have the same hparam
+	 *     setup? if so we can use best_trial_id instead of joining all trials to
+	 *     experiments in the project.
```

Comment:
The types for hyperparameters could come from the experiment config. For example, for [categorical](https://docs.determined.ai/latest/reference/reference-training/experiment-config-reference.html#categorical) hp's this could allow multiselect filter with the experiment's categorical options.

Categorical HP may have various json value type inside of it (e.g. mix of bools and numbers). If we treat "categorical" as another column type, then all trials in the experiment should have the same hyperparameter types.

This'll also require as to return the categorical categories per column in this API.

Reply:
That's a good point. One of the reasons we want the type of the hyperparameter is to give the user hints in the ui as to what a sort or filter would do (i.e. instead of showing "ascending" or "descending", we could show "A-Z" or "Z-A" for textual values and "0-9" or "9-0" for numeric values. Or, when showing the user the filter ui, we could hide filters that don't make sense for the value in question, like "contains" for textual values or "greater than/less than" for numeric values). From that standpoint, categorical hyperparameters, being non-homogeneous lists of values should be unspecified in order to make sense -- what sort example could one give for `['one', 2, ['three', 4]]`?

At any rate, it seems like sorting and filtering is operating on the experiment config and not the trial config, so this is unnecessary anyways.

Reply:
ah -- i see what you're saying.